### PR TITLE
Functional tests of yum/dnf plugins

### PR DESCRIPTION
**Pull request by jirihnidek**
* Just few Ansible playbooks for testing yum/dnf plugins.
* How to test it?
  - Edit /etc/ansible/hosts and add at least one machine to the
    group called clients. E.g.:

     [clients]
     centos7.example.com

  - Note: you have to be able to login to machines in group
    using ssh key (not password).
  - Execute:

```
     $ cd test/functional_tests/ansible_playbooks
     $ ansible-playbook ./configure_package_manager.yml \
         --extra-vars="candlepin_hostname=<IP_OF_TESTING_CANDLEPIN>"
     $ ansible-playbook ./register_system.yml
     $ ansible-playbook ./attach_subscriptions.yml
     $ ansible-playbook ./test_install_remove_packages.yml
     $ ansible-playbook ./test_not_remove_prod_cert_for_disabled_repo.yml
```
  - Note: ansible playbooks test_*.yml should pass after #2085 when
    is merged

### CODE REVIEW

**Comment by cnsnyder** [comment]
@jirihnidek Just confirming, is this ready for review? (/me wasn't sure since it was a POC)

**Comment by jirihnidek** [comment]
> @jirihnidek Just confirming, is this ready for review? (/me wasn't sure since it was a POC)

It is ready to review, but I will add instruction how to use it to some README.md and then it will let you know. :-)

**Comment by jirihnidek** [comment]
@cnsnyder I added the README.md file and I also updated one Ansible playbook.

**Comment by wottop** [comment]
Please rebase this to the current master. it is over two months stale.

**Comment by jirihnidek** [comment]
@wottop rebased and I added one more playbook to configure packaging systems (yum/dnf)

**Comment by wottop** [comment]
The final two tests hang at TASK [install packages (CentOS - yum)]

File: test/functional_tests/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  (not password).
+* Execute following commands:
+
+```bash
+cd ./ansible_playbooks
+ansible-playbook ./configure_package_manager.yml --extra-vars="candlepin_hostname=<IP_OF_TESTING_CANDLEPIN>"
+ansible-playbook ./register_system.yml
+ansible-playbook ./attach_subscriptions.yml
+ansible-playbook ./test_install_remove_packages.yml
+ansible-playbook ./test_not_remove_prod_cert_for_disabled_repo.yaml
```
Comment (wottop) [inline test/functional_tests/README.md]: .yml

File: test/functional_tests/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+centos7.example.com
+```
+
+* You have to be able to login to all machines in group using ssh key
+  (not password).
+* Execute following commands:
+
+```bash
+cd ./ansible_playbooks
+ansible-playbook ./configure_package_manager.yml --extra-vars="candlepin_hostname=<IP_OF_TESTING_CANDLEPIN>"
```
Comment (wottop) [inline test/functional_tests/README.md]: We have configured the IP of the host to 192.168.111.1 in the Vagrantfile. Perhaps just make that the default on this setup.

**Comment by wottop** [comment]
all tasks in Ansible playbook should use yum command with options: --disablerepo=* --enablerepo=<testing_repo> ...

**Comment by wottop** [comment]
Still get an error on the last two tests. Not the same errors:

http://pastebin.test.redhat.com/787696

**Comment by jirihnidek** [comment]
> Still get an error on the last two tests. Not the same errors:
> 
> http://pastebin.test.redhat.com/787696

YUM and DNF prints that testing rpms are not signed. You probably do not have installed rpm package `rpm-sign` on the machine, where candlepin was deployed. You also need following packages: `createrepo_c` or `createrepo`, `python-requests` and `rpm-build` (probably installed in your case).

**Comment by wottop** [comment]
I have installed those packages. I still see the error in the paste.

**Comment by jirihnidek** [comment]
> I have installed those packages. I still see the error in the paste.

OK and did you try to deploy candlepin server again?

What is output of following commands:

`rpm -qip /var/lib/tomcat/webapps/ROOT/snowy/foo/path/common/6051/RPMS/slow-eagle-1-07.noarch.rpm`

`rpm -K /var/lib/tomcat/webapps/ROOT/snowy/foo/path/common/6051/RPMS/slow-eagle-1-07.noarch.rpm`

Thanks.

**Comment by cnsnyder** [comment]
@jirihnidek @wottop What remains here?

**Comment by wottop** [comment]
ack, will merge when squashed.

**Pull request merged**